scalyr_agent/sharded_copying_manager/worker.py

```diff
@@ -1001,22 +991,41 @@ def __write_checkpoint_state(
         # Create the format that is expected.  An overall dict with the time when the file was written,
         # and then an entry for each file path.
         checkpoints = {}
-        state = {
-            "time": current_time,
-            "checkpoints": checkpoints,
-        }
 
         for processor in log_processors:
             if full_checkpoint or processor.is_active:
-                checkpoints[processor.log_path] = processor.get_checkpoint()
+                checkpoints[processor.get_log_path()] = processor.get_checkpoint()
 
             if full_checkpoint:
                 processor.set_inactive()
 
+        checkpoints_dir_path = os.path.join(
+            self.__config.agent_data_path, "checkpoints"
+        )
+
+        try:
+            os.mkdir(checkpoints_dir_path)
```

Comment:
Why was this code added (I believe it wasn't there originally, or was it)?

Reply:
Yes, it wasn't there originally. I've just added this to check if the `checkpoints` folder does not exist. 
But now I moved this check into the copying manager, now it does the check when it combines all worker checkpoint files into one master file.